### PR TITLE
Attach cleanup listeners to ensure the temp file is cleaned up properly

### LINKED ISSUE #27: File can be left over if process is terminated before cleanup happens
Problem: the process might be terminated before the temporary config cleanup is done

https://github.com/gustavopch/tsc-files/blob/27a82464f2c0ff26ccb5b5b152b589aa580bd5a6/lib/index.js#L54

When that happens - and I've had that happen multiple times e.g. when hitting `Ctrl-C` (`SIGINT`) on a parent process - the file will not be cleaned up properly.

Solution: attach listeners on termination signals for cleaning up the file when the process exits (https://nodejs.org/api/process.html#signal-events)

### PR DESCRIPTION
**Pull request by resolritter**
close #27

### CODE REVIEW

**Comment by gustavopch** [comment]
@resolritter Thanks for the PR. I want to make some code-style changes. Can you enable edits? https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/working-with-forks/allowing-changes-to-a-pull-request-branch-created-from-a-fork

**Comment by resolritter** [comment]
@gustavopch done

**Comment by gustavopch** [comment]
@didaktio I'm really sorry, but I'm not having time to dedicate to this repo as I'm already overwhelmed with work stuff, and I'm not even using TypeScript these days. I also don't think merging without testing would be a good idea as it could create still more maintenance needs in case it breaks something. The whole thing is less than 100 LoC, so I encourage people to fork or maybe just copy the code into their projects and adapt as needed.

File: lib/index.js [hunk 1]
```diff
@@ -36,10 +39,63 @@ const tmpTsconfig = {
...
+    exitCallback(
+      signal,
+      /*
+        The "exit" signal is emitted when Node.js is exiting normally, thus
+        (shouldExit = true) is not necessary in that case
+      */
+      signal === 'exit' ? false : true
+    )
+  )
+}
```
Comment (fearphage) [inline lib/index.js:97]: This logic can be simplified.

```suggestion
[
  'exit',
  'SIGINT',
  'SIGTERM',
  'SIGQUIT',
  'SIGHUP',
  'SIGPIPE',
].forEach(signal => process.on(
  signal, 
  exitCallback(
    signal,
    /*
      The "exit" signal is emitted when Node.js is exiting normally, thus
      (shouldExit = true) is not necessary in that case
    */
    signal !== 'exit',
  )
)
```

File: lib/index.js [hunk 2]
```diff
@@ -36,10 +39,63 @@ const tmpTsconfig = {
...
+]) {
+  process.on(
+    signal,
+    exitCallback(
+      signal,
+      /*
+        The "exit" signal is emitted when Node.js is exiting normally, thus
+        (shouldExit = true) is not necessary in that case
+      */
+      signal === 'exit' ? false : true
```
Comment (fearphage) [inline lib/index.js:94]: This logic can be simplified. You should rarely return an `boolean` from a `boolean` expression when you can return the value of the `boolean` expression.

```suggestion
      signal !== 'exit'
```

**Comment by gustavopch** [comment]
Implemented in https://github.com/gustavopch/tsc-files/releases/tag/v1.1.4. Thanks for the PR and sorry for the delay.

**Pull request closed without merging**